client/app/scripts/superdesk-vocabularies/vocabularies.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * This file is part of Superdesk.
+ *
+ * Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code, or
+ * at https://www.sourcefabric.org/superdesk/license
+ */
+
+(function() {
+    'use strict';
+
+    VocabularyService
+        .$inject = [ 'api', 'urls', 'session', '$q'];
+    function VocabularyService(api, urls, session, $q) {
+        var service = this;
+
+        this.getVocabularies = function() {
+            if (typeof service.vocabularies === 'undefined') {
+                return api.query('vocabularies')
+                .then(function(result) {
+                    service.vocabularies = result;
+                    return $q.when(service.vocabularies);
+                });
+            } else {
+                return $q.when(service.vocabularies);
+            }
+        };
```

Comment:
Sorry, my bad, should have worded it more clearly - _some_ tests are missing. The linked test case tests that the method makes a correct API call, but there should also be other test cases, testing e.g. whether the returned promise is resolved with a correct value and that the list of vocabularies is stored in the service itself.

BTW, why is a scope object passed to the `getVocabularies()` method in the linked test, the method does not have any parameters?